server/container/guice/protocols/jmap/src/main/java/org/apache/james/utils/MessageIdProbe.java

```diff
@@ -47,4 +51,10 @@ public MessageIdProbe(MailboxManager mailboxManager, MessageIdManager messageIdM
 
         return messageIdManager.getMessages(ImmutableList.of(messageId), FetchGroupImpl.FULL_CONTENT, mailboxSession);
     }
+
+    public void setFlag(String user, Flags newFlags, MessageId messageId, List<MailboxId> mailboxIds) throws MailboxException {
+        MailboxSession mailboxSession = mailboxManager.createSystemSession(user);
+
+        messageIdManager.setFlags(newFlags, FlagsUpdateMode.ADD, messageId, mailboxIds, mailboxSession);
```

Comment:
A method called ``setFlag`` make me think you actually want to replace old set of flags by the new one. But here you use ``FlagsUpdateMode.ADD``. What do you really want, if you want ``ADD``, I think we should found a more explicit name for this method for example I would suggest ``addFlag``

Reply:
I changed the method name and parameter value